hybridse/include/codec/list_iterator_codec.h

```diff
@@ -547,15 +526,90 @@ class ColumnIterator : public ConstIterator<uint64_t, V> {
         value_ = column_impl_->GetFieldUnsafe(row_iter_->GetValue());
         return value_;
     }
+    bool IsValueNull() override {
+        return column_impl_->IsNull(row_iter_->GetValue());
+    }
     const uint64_t &GetKey() const override { return row_iter_->GetKey(); }
     bool IsSeekable() const override { return row_iter_->IsSeekable(); }
 
- private:
-    const ColumnImpl<V> *column_impl_;
+ protected:
     std::unique_ptr<RowIterator> row_iter_;
+    const ColumnImpl<V> *column_impl_;
     V value_;
 };
 
+template <class V>
+class NonNullColumnIterator : public ColumnIterator<V> {
+ public:
+    NonNullColumnIterator(std::unique_ptr<RowIterator> &&it, const ColumnImpl<V> *column_impl)
+    : ColumnIterator<V>(std::move(it), column_impl), valid_(true) {
+        NextNonNull();
+    }
+    ~NonNullColumnIterator() override {}
+
+    void Seek(const uint64_t &key) override {
+        valid_ = true;
+        this->row_iter_->Seek(key);
+        NextNonNull(true);
+    }
+
+    void SeekToFirst() override {
+        valid_ = true;
+        this->row_iter_->SeekToFirst();
+        NextNonNull();
+    }
+    bool Valid() const override { return valid_ && this->row_iter_->Valid(); }
+
+    void Next() override {
+        this->row_iter_->Next();
+        NextNonNull();
+    }
+
+ private:
+    void NextNonNull(bool current_key_only = false) {
+        if (!this->row_iter_ || !this->row_iter_->Valid()) {
+            return;
+        }
+
+        auto key = this->row_iter_->GetKey();
+
+        while (this->row_iter_->Valid()) {
+            bool is_null = true;
+            V val {};
+            this->column_impl_->GetField(this->row_iter_->GetValue(), &val, &is_null);
+            if (!is_null) {
+                break;
+            }
+
+            if (current_key_only && key != this->row_iter_->GetKey()) {
+                // invalid if key not seeked
+                valid_ = false;
```

Comment:
seeking to a key with all values `NULL` invalids iterator, means seeking failed. see if this behavior ok .